test/react-web/client/graphql/queries.test.tsx

```diff
@@ -2113,4 +2113,39 @@ describe('queries', () => {
     );
   });
 
+  it('uses a custom wrapped component name when \'alias\' is specified', (done) => {
+    const query = gql`query people { allPeople(first: 1) { people { name } } }`;
+    const data = { allPeople: { people: [ { name: 'Luke Skywalker' } ] } };
+    const networkInterface = mockNetworkInterface({ request: { query }, result: { data } });
+    const client = new ApolloClient({ networkInterface, addTypename: false });
+
+    let queryExecuted;
+    @graphql(query, {
+      alias: 'withFoo',
+    })
+    class Container extends React.Component<any, any> {
+      componentWillReceiveProps(props) {
```

Comment:
I get an error: `Invariant Violation: Could not find "client" in the context of "withFoo(Container)". Wrap the root component in an <ApolloProvider>`. 

But if I wrap the component in `<ApolloProvider>`, I'm not sure how to access its `displayName`…